extensions/sockets/src/main/java/org/mule/module/socket/api/socket/SocketProperties.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.socket.api.socket;
+
+/**
+ * Interface for objects that provide common configuration for generic sockets.
+ *
+ * @since 4.0
+ */
+public interface SocketProperties
+{
+
+    /**
+     * The size of the buffer (in bytes) used when sending data, set on the socket itself.
+     */
+    Integer getSendBufferSize();
```

Comment:
does it actually makes sense for these methods to return `Integer`? Can any of these be null? Seems weird to me since you always mention defaults in the javadocs. If there's no reason why these couldn't be `int` instead then change it and make sure that all the places where these methods are use an `int` is also expected